### PR TITLE
[MAJOR] Class based elements

### LINKED ISSUE #233: Provide non Mixin approach
In Ember Octane, the `Router` class is generated as a native class. 

Because it's still based on an Ember core object, mixing in RouterScroll still works:

```js
import EmberRouter from '@ember/routing/router';

export default class Router extends EmberRouter.extend(RouterScroll) {
}
```

but in the long term, it would be good to remove the dependence on the Ember object model and provide a different way to provide this feature.

### PR DESCRIPTION
**Pull request by snewcomer**
The proposal here would replace our `Mixin` with a Class based approach.

```js
class EmberRouterScroll extends EmberRouter {}
```

^^ this is imported in your app.

This doesn't seem very flexible now that I have this pushed up.  However, single inheritance is biting me at the moment.  Will think about this further unless other's have thoughts!

This would be a BREAKING change.

close #233 

@mehulkar

### CODE REVIEW

**Comment by chriskrycho** [comment]
One of the possible tricks with single inheritance – any user would need to further extend from *this*. I've been mulling on alternative approaches here, including just having users explicitly opt in via the appropriate hook or *maaaaybe* (and I'm very iffy on this) a class decorator.

File: package.json [hunk 1]
```diff
@@ -53,40 +53,42 @@
...
-    "ember-load-initializers": "^2.0.0",
+    "ember-load-initializers": "^2.1.0",
     "ember-maybe-import-regenerator": "^0.1.6",
-    "ember-qunit": "^4.4.1",
-    "ember-resolver": "^5.0.1",
-    "ember-source": "~3.12.0",
-    "ember-source-channel-url": "^1.1.0",
+    "ember-qunit": "^4.5.1",
+    "ember-resolver": "^5.3.0",
+    "ember-source": "~3.14.1",
```
Comment (snewcomer) [inline package.json]: Will do Octane after this PR.

File: addon/index.js [hunk 2]
```diff
@@ -46,32 +45,32 @@ function tryScrollRecursively(fn, scrollHash) {
   })
 }
 
-let RouterScrollMixin = Mixin.create({
-  service: inject('router-scroll'),
+class EmberRouterScroll extends EmberRouter {
```
Comment (snewcomer) [inline addon/index.js]: Although this limits some flexibility, we will follow up with util functions that you can import and call in init to setup listeners as well if this pattern does not fix your app.

Comment (snewcomer) [reply addon/index.js]: This pattern - meaning extending from `EmberRouter`

**Pull request merged**